crossdock/client/outboundttl/behavior.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package outboundttl
+
+import (
+	"strings"
+	"time"
+
+	"github.com/yarpc/yarpc-go/crossdock-go"
+	"github.com/yarpc/yarpc-go/crossdock/client/rpc"
+	"github.com/yarpc/yarpc-go/encoding/raw"
+	"github.com/yarpc/yarpc-go/transport"
+
+	"golang.org/x/net/context"
+)
+
+func Run(t crossdock.T) {
+	assert := crossdock.Assert(t)
+	fatals := crossdock.Fatals(t)
+
+	rpc := rpc.Create(t)
+	ch := raw.New(rpc.Channel("yarpc-test"))
+	_, _, err := ch.Call(&raw.ReqMeta{
+		Context:   newTestContext(),
+		Procedure: "sleep/raw",
+		TTL:       time.Millisecond * 100,
+	}, make([]byte, 0))
+	fatals.Error(err, "expected a failure for timeout")
+
+	form := strings.HasPrefix(err.Error(), "timeout for procedure \"sleep/raw\" of service \"yarpc-test\" after")
+	assert.True(form, "error message has expected prefix for timeouts, got %q", err.Error())
+	_, ok := err.(transport.TimeoutError)
+	assert.True(ok, "transport should be a TimeoutError")
```

Comment:
I expect that comparing a first-class TimeoutError instance against the error would fail for instances with the TimeoutError behavior that are not exactly instances of the hidden timeoutError struct type. Trying to assert the behavior http://dave.cheney.net/2014/12/24/inspecting-errors